flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/validation/DefaultValidator.java

```diff
@@ -58,6 +58,8 @@ public class DefaultValidator implements FlinkResourceValidator {
     @Override
     public Optional<String> validateDeployment(FlinkDeployment deployment) {
         FlinkDeploymentSpec spec = deployment.getSpec();
+        Map<String, String> effectiveConfig =
+                FlinkUtils.getFlinkEffectiveConfig(deployment).toMap();
```

Comment:
I think this is still incorrect from a validation perspective ,we should simply make a copy from the default config and put all values.:

```
Configuration conf = new Configuration(defaultConf)
conf.putAll(spec.getFlinkConfiguration)
```